### ISSUE TITLE
Thresholding for embeddings -BERT

### ISSUE THREAD

**Issue by Aditi00a** [JohnSnowLabs/spark-nlp#6507]
**Is your feature request related to a problem? Please describe.**
I have an NER pipeline where I am retraining using custom labels. Prior to this using BERT EMBEDDINGS
`        
embeddings = BertEmbeddings.pretrained("bert_base_cased", 'en'). \
            setInputCols(["sentence", 'token']). \
            setOutputCol("embeddings")

loaded_ner_model = NerDLModel.load(model_file) \
        .setInputCols(["sentence", "token", "embeddings"]) \
        .setOutputCol("ner")
`
Is there any way we can add a parameter to add a thresholding for the "ner" generated ?
Also, is this a multiclass classification ?if yes, is there any way to change the final softmax thresholding for each class ?

**Describe the solution you'd like**
Add a parameter to add a thresholding for the "ner" generated. 
Way to change the final softmax thresholding for each class ?

**Comment by maziyarpanahi** [comment]
Hi,

The NerDL (NerDLModel - NerDLApproach) is a multi-label token classification. That's being said, by threshold you mean if the confidence score is not above a certain point (set by user) it should not return anything as a label? (or return `O` or return a label also set by the user?) 

The reason I am asking is that the confidence score is turned off by default to optimize the speed. (less info being fetched from TF graph the better). So this only works if users set this to True `setIncludeConfidence` in NerDLModel. We also have a similar feature for other multi-class annotators if you want to look at it as a reference to what I asked: https://nlp.johnsnowlabs.com/api/com/johnsnowlabs/nlp/annotators/classifier/dl/SentimentDLModel.html#threshold:org.apache.spark.ml.param.FloatParam

**Comment by Aditi00a** [comment]
yes, this is my requirement - if the confidence score is not above a certain point (set by user) it should not return anything as a label? (or return O or return a label also set by the user?)

so, after setting setIncludeConfidence in NerDLModel as True, can i set that "certain point" in the pipeline ?

**Comment by maziyarpanahi** [comment]
Yes, if you set `setIncludeConfidence` to True for each prediction inside the .metadata you can see the score. I suggest having a look there to get a better idea of the scores and then we can come back to see if having `threshold` can help here.

**Comment by maziyarpanahi** [comment]
Hi,

1. it will already predict the label with the highest confidence score
2. no, it's already predicted when it comes out of the TensorFlow, we only see the label and its score. If you wish to filter some predictions out you can manually to those manipulations via a UDF.
